Kamu UI 136 refactor observable subscriptions to pipe async

dmitriy-borzenko opened the pull request.

zaychenko-sergei commented on src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts:
```diff
@@ -11,25 +11,26 @@ import {
 })
-export class EventDetailsComponent extends BaseComponent implements OnInit {
+export class EventDetailsComponent {
     public block: MetadataBlockFragment;
```
I suggest storing `block` field in the `BlockService` instead.
You could simply make a "currentBlock" property to access from component, as you don't need it in your template, only in methods.

dmitriy-borzenko replied:
Moved `block` to `BlockService `

zaychenko-sergei commented on src/app/dataset-block/metadata-block/components/yaml-view-section/yaml-view-section.component.ts:
```diff
@@ -12,8 +13,8 @@ import { SupportedEvents } from "../event-details/supported.events";
 })
 export class YamlViewSectionComponent extends BaseComponent implements OnInit {
     public block: MetadataBlockFragment;
```
Similarly, let's move the `block` to `BlockService`

dmitriy-borzenko replied:
Moved block to BlockService

zaychenko-sergei commented on src/app/auth/account/account.component.ts:
```diff
@@ -23,14 +23,12 @@ import { filter, map } from "rxjs/operators";
 export class AccountComponent extends BaseComponent implements OnInit {
```
Change detector left unused

dmitriy-borzenko replied:
Deleted

EvilAlexei commented on src/app/auth/account/account.component.html:
```diff
@@ -125,10 +127,10 @@ <h2 class="text-start pt-3 px-3 h6 organization">Organizations</h2>
             <ng-container *ngIf="isAccountViewTypeDatasets">
                 <app-datasets-tab
-                    [datasets]="datasets"
+                    [datasets]="datasets.datasets"
```
I suggest to rename datasets, for example to `data`, to avoid name duplication

dmitriy-borzenko replied:
Ok, renamed

EvilAlexei commented on src/app/auth/account/account.component.ts:
```diff
@@ -1,6 +1,5 @@
-import { DatasetSearchOverviewFragment, PageBasedInfo } from "./../../api/kamu.graphql.interface";
 import { BaseComponent } from "src/app/common/base.component";
 import { NavigationService } from "src/app/services/navigation.service";
 import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, OnInit, ViewChild } from "@angular/core";
```
And I think you don't need `ChangeDetectorRef` and I can be removed

dmitriy-borzenko replied:
Fixed

EvilAlexei commented on src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts:
```diff
@@ -11,25 +11,26 @@ import {
-export class EventDetailsComponent extends BaseComponent implements OnInit {
+export class EventDetailsComponent {
     public block: MetadataBlockFragment;
+    public block$: Observable<MetadataBlockFragment> = this.blockService.onMetadataBlockChanges.pipe(
+        tap((block: MetadataBlockFragment) => (this.block = block)),
+    );
```
I suggest moving this to the constructor, for readability and better structuring,
I mean assigning the Observable.

My thoughts:
We will have a section to create parameters,
and we will have a section with initialization and some settings

dmitriy-borzenko replied:
Ok

EvilAlexei commented on src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts:
```diff
@@ -11,116 +11,109 @@ import {
-export class EventDetailsComponent extends BaseComponent implements OnInit {
-    public block: MetadataBlockFragment;
+export class EventDetailsComponent {
+    public block$: Observable<MetadataBlockFragment> = this.blockService.onMetadataBlockChanges.pipe();
```
I think you don't need a `pipe` here.

dmitriy-borzenko replied:
Fixed

dmitriy-borzenko commented:
> I've added a more complex example in `MetadataBlockComponent`. Please review more candidates where `cdr` could be removed / replaced with pipes:
> 
>     * `DataComponent`
> 
>     * `DatasetNameByIdPropertyComponent`
> 
>     * `ngOnInit` in `BaseStepComponent` (don't we have a good auto refresh??)



> I've added a more complex example in `MetadataBlockComponent`. Please review more candidates where `cdr` could be removed / replaced with pipes:
> 
>     * `DataComponent`
> 
>     * `DatasetNameByIdPropertyComponent`
> 
>     * `ngOnInit` in `BaseStepComponent` (don't we have a good auto refresh??)

Fixed
